src/app.js

```diff
@@ -222,15 +238,44 @@ class SAFEApp extends EventEmitter {
           file = await emulation.fetch(filePath);
         }
         const openedFile = await emulation.open(file, consts.pubConsts.NFS_FILE_MODE_READ);
-        const data = await openedFile.read(
-          consts.pubConsts.NFS_FILE_START, consts.pubConsts.NFS_FILE_END);
+        let range;
+        let start = consts.pubConsts.NFS_FILE_START;
+        let end;
+        let fileSize;
+        let fileFinalByteIndex;
+        let lengthToRead = consts.pubConsts.NFS_FILE_END;
+        let endByte;
+
+        // TODO: how do we handle multipart Reqs
+        if (options && options.range && typeof options.range.start !== 'undefined') {
+          range = options.range;
+          start = range.start;
+          fileSize = await openedFile.size();
+          end = range.end || fileSize;
+          fileFinalByteIndex = fileSize - 1;
+          lengthToRead = end - start;
+
+          if (end === fileFinalByteIndex) {
+            lengthToRead += 1;
+          }
+        }
+        const data = await openedFile.read(start, lengthToRead);
         const mimeType = mime.getType(nodePath.extname(filePath)) || 'application/octet-stream';
-        resolve({
+
+        const response = {
           headers: {
             'Content-Type': mimeType
           },
           body: data
-        });
+        };
+
+        if (range) {
+          endByte = (end === fileSize) ? fileSize - 1 : end;
+          response.headers['Content-Range'] = `bytes ${start}-${endByte}/${fileFinalByteIndex}`;
```

Comment:
I think the value after the `/` is incorrect, it should be the length not the last byte index